### ISSUE TITLE
Ligatures on [INFO] don't seem to work with coloured output

### ISSUE THREAD

**Issue by talios** [fabrizioschiavi/pragmatapro#40]
Loving the new [INFO] ligatures, but it seems they don't always seem to fire.  Apache Maven (3.4-SNAPSHOT shown here ) recently introduced coloured output arround the [INFO] [ERROR] blocks, which don't seem to like the ligature support.

<img width="290" alt="colors-and-ligatures" src="https://cloud.githubusercontent.com/assets/19093/18619814/ea4d2c5e-7e59-11e6-9c45-1898239dca50.png">

Is this something that can be solved in the font itself?

**Comment by fabrizioschiavi** [comment]
Thanks for the notification, I'll investigate about this case but I already did all I could do from the “font-side”

**Comment by tiziano88** [comment]
I guess that's because the color change breaks it into multiple segments (e.g. `[\e[32mINFO\e[39m]` (unescaped)). Anyway I don't think there is anything to be done on the font side, for instance what colour would the ligature be if the underlying text was made up of several colours?

**Comment by freebroccolo** [comment]
Indeed, this is the same problem that shows up in editors where sometimes syntax highlighters will tokenize the input in a way that splits up adjacent characters needed to form the ligature, causing them to not render as expected.

What you could do here is strip the color (at least around where the ligature should be) using some sort of post-processing. Not ideal but probably the only realistic option unless you wanted to disable color globally.

I think the stylistic set approach might have solved this but doing it that way has other downsides, like being harder to use in most programs.

**Comment by Porges** [comment]
Could the font ignore the control characters? Or would this require N ligatures, one per colour?

**Comment by TyLeRRR** [comment]
Hi folks. Facing here some similar issue with Maven+ITerm2+PragmataPro (0.827).
Not sure whether this is for a separate ticket.

<img width="257" alt="Screenshot 2019-03-17 at 12 31 33" src="https://user-images.githubusercontent.com/7289706/54489844-feee8800-48b0-11e9-9345-76c5348c830e.png">

Have no problems with these ligatures ([INFO], [WARNING], etc.) in IntelliJ IDEA terminal, but in ITerm2 they refuse to work. Any suggestions? Am I doing smth wrong?

**Comment by Bobbbay** [comment]
Doesn't work on alacritty either :/
